File: sample/aforth/benchmark/bm_nested_loop.fs
\ Nested DO loop — pure dispatch + counter arith, no allocations.
\ Computes sum_{i=0..N-1} sum_{j=0..N-1} (i*j).

: matsum ( n -- s )
  0 SWAP                        \ ( accum n )
  DUP 0 DO
    DUP 0 DO
      I J * ROT + SWAP          \ accum += i*j
    LOOP
  LOOP
  DROP ;

8000 matsum . CR

File: sample/koruby_precise/benchmark/bm_nested_loop.rb
# Nested loop (integer arithmetic)
def bench
  sum = 0
  i = 0
  while i < 200
    j = 0
    while j < 200
      sum += i * j
      j += 1
    end
    i += 1
  end
  sum
end

result = 0
i = 0
while i < 1000
  result = bench
  i += 1
end
p(result)
